ports/esp32s2/common-hal/alarm/__init__.c

```diff
@@ -110,6 +115,17 @@ STATIC void _setup_sleep_alarms(bool deep_sleep, size_t n_alarms, const mp_obj_t
             }
             time_alarm  = MP_OBJ_TO_PTR(alarms[i]);
             time_alarm_set = true;
+        } else if (MP_OBJ_IS_TYPE(alarms[i], &alarm_touch_touchalarm_type)) {
+            if (!touch_alarm_set) {
+                if (deep_sleep) {
+                    alarm_touch_touchalarm_set_alarm(MP_OBJ_TO_PTR(alarms[i]));
+                    touch_alarm_set = true;
+                } else {
+                    mp_raise_NotImplementedError(translate("TouchAlarm not available in light sleep"));
+                }
+            } else {
+                mp_raise_ValueError(translate("Only one alarm.touch alarm can be set."));
+            }
```

Comment:
Please factor this out into the touch alarm code. (See my pin alarm PR for how I've done it there.)